extensions/ql-vscode/src/query-history.ts

```diff
@@ -59,16 +59,15 @@ interface QueryHistoryDataProvider extends vscode.TreeDataProvider<CompletedQuer
 /**
  * Tree data provider for the query history view.
  */
-class HistoryTreeDataProvider implements QueryHistoryDataProvider {
+class HistoryTreeDataProvider extends DisposableObject implements QueryHistoryDataProvider {
```

Comment:
One issue we were having during integration testing was the fact that some things were not properly being disposed. I went through and made sure more things had proper disposal. So, you'll see that spread throughout this commit.